tests/conftest.py

```diff
@@ -159,6 +174,15 @@ def pytest_configure(config):
     config.addinivalue_line(
         "markers", "runtime(name): mark test to run with named runtime"
     )
+    config.addinivalue_line(
+        "markers", "system(name): mark test to run with named system"
+    )
+    config.addinivalue_line(
+        "markers", "long_lasting(name): mark test that run for a long time (many minutes)"
+    )
+    config.addinivalue_line(
+        "markers", "credential_file(name): mark tests that require credential file in CREDENTIALS_DIR"
```

Comment:
Does it make sense for credential_file to be a "marker" in the pytest-sense, This isn't something I'd imagine we'd select by on the command line (but system etc do make sense as markers).

Are these perhaps instead just a property that should be set on the SystemTest sub-class?

Reply:
It makes sense because they also determine whether the test is skipped or not - if the file with credential is missing, then the test is automatically skipped (and reason for skipping explains what credential is missing). I think it's generic enough to be used in a number of cases (and we can include other markers for that - credential_variable for example for env variables missing). The point here is to skip the tests when the credentials are missing.